fix: quick terminal `focus-follows-mouse` behaviour

dmehala opened the pull request:
# Description

Quick Terminal now focuses on the surface under the mouse pointer when `focus-follows-mouse` is enabled.

Fixes #3337

mitchellh commented on macos/Sources/Features/QuickTerminal/QuickTerminalWindow.swift:
```diff
@@ -1,6 +1,8 @@
 import Cocoa
 
 class QuickTerminalWindow: NSWindow {
+    var focusFollowsMouse: Bool = false
```
I think we should add this to `BaseTerminalController` and use that instead of a window property. That would simplify everything else in SurfaceView if we introduce new terminal types.

mitchellh commented on macos/Sources/Ghostty/SurfaceView_AppKit.swift:
```diff
@@ -617,12 +617,19 @@ extension Ghostty {
-                window.focusFollowsMouse &&
-                !self.focused
-            {
+            if (self.window!.isKeyWindow && !self.focused && focusFollowsMouse) {
```
I'd prefer to guard against window being null here. It is possible (we detach views at certain times).

mitchellh commented on macos/Sources/Features/QuickTerminal/QuickTerminalController.swift:
```diff
@@ -67,6 +67,9 @@ class QuickTerminalController: BaseTerminalController {
         window.isRestorable = false
+        
+        // Propagate `focus-follows-mouse` options to the quick terminal.
+        window.focusFollowsMouse = ghostty.config.focusFollowsMouse
```
BaseTerminalController has a windowDidLoad you can add this to. It might make sense to add a function there to call on config change too (notification is already there too).

mitchellh commented:
Appreciate this. 😄 Suggested some changes.

Commit:
fix: quick terminal `focus-follows-mouse` behaviour

Quick Terminal now focuses on the surface under the mouse
pointer when `focus-follows-mouse` is enabled.

Fixes #3337

dmehala commented:
I completely reverted https://github.com/dmehala/ghostty/commit/0ec402f926c66b2c94e217c18a30f55e36a5fe02 and addressed your comments. Now `BaseTerminalController` have a new `followFocusMouse` property which is used by `SurfaceView`. I'm not a Swift or AppKit expert, so this might not be the best approach 😅, but please feel free to take over if that works better for you.

dmehala commented on macos/Sources/Features/Terminal/BaseTerminalController.swift:
```diff
@@ -62,6 +62,8 @@ class BaseTerminalController: NSWindowController,
 
     /// The configuration derived from the Ghostty config so we don't need to rely on references.
     private var derivedConfig: DerivedConfig
+    
+    private(set) var followFocusMouse: Bool = false
```
```suggestion
    /// Keep track of whether the controller should focus on the pane currently being hovered.
    private(set) var followFocusMouse: Bool = false
```

mitchellh commented:
Works great. Made some stylistic changes feel free to take a look. Will merge once green. Thanks
